pipeline.go

```diff
@@ -95,7 +96,15 @@ func clearSetup() error {
 	return nil
 }
 
-// Run executes the pipeline.
+// Run executes the pipeline. For each stage defined in the pipeline we execute
+// the `docker build` and `docker run`. If any of these two processes fail,
```

Comment:
Vale a pena definir o que queremos dizer com erro. Nesse caso, um erro de execução ou o processamento do run retorna um código de status diferente de 0

Reply:
Certo. Adicionei o seguinte:
"Here we consider a failure when the building or execution of the image returns a status other than 0 or when an error is raised within the buildImage or runImage functions."

Reply:
Ainda senti falta de uma coisinha: deixar claro qual a stdin que o ErrorHandler é chamado.

Reply:
Certo. Vou fazer o merge dessa PR e abrir uma outra porque acabei melhorando também outras descrições e ai fica melhor de visualizar.